components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/ext/listener/ExtensionListenerUtil.java

```diff
@@ -264,10 +264,18 @@ private static boolean handleExtensionResponse(MessageContext messageContext,
         }
         // handle payload
         processResponsePayload(extensionResponseDTO, messageContext);
+        //Extract customProperty from DTO
+        Map<String, Object> customProperty = extensionResponseDTO.getCustomProperty();
+        //set Additional analytics Properties to the message context.
+        if(customProperty != null && customProperty.get(APIMgtGatewayConstants.ADDITIONAL_ANALYTICS_PROPS) != null ) {
+            Map<String, Object> customElkAttributes =
+                    (Map<String, Object>) customProperty.get(APIMgtGatewayConstants.ADDITIONAL_ANALYTICS_PROPS);
```

Comment:
Do we really need to do the casting to `Map<String, Object>` here as we are just setting it back to the `messageContext` as an object?

Reply:
fixed with https://github.com/wso2/carbon-apimgt/pull/11772/commits/275e724ab7f475240e8ccb51f6d360e6c12ca298